linalg/fdsolver.hpp

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2010-2021, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#ifndef MFEM_FDSOLVER
+#define MFEM_FDSOLVER
+
+#include "../config/config.hpp"
+#include "densemat.hpp"
+
+namespace mfem
+{
+
+/// Computes the inverse diagonal dinv = (a⊗I + I⊗b)^-1
+/// where a, b are diagonal matrices and I is the identity of the
+/// appropriate size
+void KronMultInvDiag(const Vector & a, const Vector & b, Vector & dinv);
+
+/// Computes the inverse diagonal dinv = (a⊗I⊗I + I⊗b⊗I + I⊗I⊗c)^-1
+/// where a, b, c are diagonal matrices and I is the identity of the
+/// appropriate size
+void KronMultInvDiag(const Vector & a, const Vector & b,
+                     const Vector & c, Vector & dinv);
+
+void KronMultInvDiag(const Array<Vector *> & X, Vector & diag);
```

Comment:
Maybe `KronMult` is the wrong term here, since these construct the diagonal rather than multiply by it. Perhaps `KronProdInvDiag`?